src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectProtocolValidator.cs

```diff
@@ -48,24 +48,14 @@ namespace Microsoft.IdentityModel.Protocols.OpenIdConnect
     /// <see cref="OpenIdConnectProtocolValidator"/> is used to ensure that an <see cref="OpenIdConnectMessage"/>
     ///  obtained using OpenIdConnect is compliant with  http://openid.net/specs/openid-connect-core-1_0.html .
     /// </summary>
-    public class OpenIdConnectProtocolValidator
+    public class OpenIdConnectProtocolValidator : IDisposable
     {
-        private IDictionary<string, string> _hashAlgorithmMap =
-            new Dictionary<string, string>
-            {
-                { SecurityAlgorithms.EcdsaSha256, "SHA256" },
-                { SecurityAlgorithms.RsaSha256, "SHA256" },
-                { SecurityAlgorithms.HmacSha256, "SHA256" },
-                { SecurityAlgorithms.EcdsaSha384, "SHA384" },
-                { SecurityAlgorithms.RsaSha384, "SHA384" },
-                { SecurityAlgorithms.HmacSha384, "SHA384" },
-                { SecurityAlgorithms.EcdsaSha512, "SHA512" },
-                { SecurityAlgorithms.RsaSha512, "SHA512" },
-                { SecurityAlgorithms.HmacSha512, "SHA512" },
-          };
+        private IDictionary<ISet<string>, HashAlgorithm> _hashAlgorithmMap;
```

Comment:
We moved away from the pattern where we had local privates for public properties. Should we just have the `HashAlgorithm` property?